drivers/led/led_gpio.c

```diff
@@ -19,24 +20,33 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_LOG_LEVEL);
 
+struct led_gpio {
+	size_t num_gpios;
+	const struct gpio_dt_spec *gpio;
+};
+
 struct led_gpio_config {
 	size_t num_leds;
-	const struct gpio_dt_spec *led;
+	const struct led_gpio *led;
 };
 
 static int led_gpio_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
 {
 
 	const struct led_gpio_config *config = dev->config;
-	const struct gpio_dt_spec *led_gpio;
+	const struct led_gpio *led_gpio;
 
 	if ((led >= config->num_leds) || (value > 100)) {
 		return -EINVAL;
 	}
 
 	led_gpio = &config->led[led];
 
-	return gpio_pin_set_dt(led_gpio, value > 0);
+	if (led_gpio->num_gpios != 1) {
+		return -ENOSYS;
```

Comment:
I would change that to either `EINVAL` or `ENOTSUP`.

Reply:
Well, I thought the same thing but `ENOTSUP` stands for "Unsupported value" and `EINVAL` for "Invalid argument". This doesn't fit very well. `ENOSYS` stands for "Function not implemented" and is used in `include/zephyr/drivers/led.h` when an API method is not implemented. That's why I chose it. Let me know what you think.

Reply:
IMO. we should not use `ENOSYS` here. Since `num_gpios` member is not passed by a function argument, I would use `ENOTSUP` errno which precisely tells you that this value is not supported.

Reply:
It "kind of" makes sense :) I changed it to `ENOTSUP`.